crates/corro-agent/src/api/peer.rs

```diff
@@ -215,7 +215,10 @@ async fn build_quinn_server_config(config: &GossipConfig) -> eyre::Result<quinn:
 pub async fn gossip_server_endpoint(config: &GossipConfig) -> eyre::Result<quinn::Endpoint> {
     let server_config = build_quinn_server_config(config).await?;
 
-    Ok(quinn::Endpoint::server(server_config, config.bind_addr)?)
+    Ok(quinn::Endpoint::server(
+        server_config,
+        config.external_addr.unwrap_or(config.bind_addr),
```

Comment:
I think we always want to bind to `config.bind_addr`!

That logic should be used here instead (maybe with a new `Agent` function for it): https://github.com/superfly/corrosion/blob/6d1b5ecc890a16963209b103b605f0d172f4d52c/crates/corro-agent/src/agent.rs#L442